tests/capi/unittest_capi_inference_nnfw_runtime.cc

```diff
@@ -15,87 +15,190 @@
-      g_mutex_lock (&g_test_mutex);                   \
-      done = (*sink_called_cnt >= expected_cnt);      \
-      waiting_time += 10000U;                         \
-      g_mutex_unlock (&g_test_mutex);                 \
-      if (!done)                                      \
-        g_usleep (10000U);                            \
-    }                                                 \
-    ASSERT_TRUE (done);                               \
-  } while (0)
+static GMutex g_test_mutex;   /** Global mutex lock for multiple sink callback */
+guint *g_sink_called_cnt;     /** Global counter for multiple sink callback */
 
 /**
- * @brief Get model file after validation checks
- * @returns model file path, NULL on error
- * @note caller has to be free the returned model file path
+ * @brief Test Fixture class for ML API of the NNFW Inference
  */
-static gchar *
-get_model_file ()
+class MLAPIInferenceNNFW : public ::testing::Test
 {
-  gchar *model_file;
-  gchar *meta_file;
-  gchar *model_path;
-  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
-
-  /* supposed to run test in build directory */
-  if (root_path == NULL)
-    root_path = "..";
-
-  /** nnfw needs a directory with model file and metadata in that directory */
-  model_path = g_build_filename (root_path, "tests", "test_models", "models", NULL);
-
-  meta_file = g_build_filename (model_path, "metadata", "MANIFEST", NULL);
-  if (!g_file_test (meta_file, G_FILE_TEST_EXISTS)) {
-    g_free (model_path);
-    g_free (meta_file);
-    return NULL;
+protected:
+  ml_single_h single_h;
+  ml_pipeline_h pipeline_h;
+  ml_tensors_info_h in_info, out_info;
+  ml_tensors_info_h in_res, out_res;
+  ml_tensor_dimension in_dim, out_dim, res_dim;
+  ml_tensors_data_h input, input2, output;
+  const gchar *root_path;
+  const gchar *valid_model;
+
+  /**
+   * @brief Get the valid model file for NNFW test
+   * @return gchar* the path of the model file
+   */
+  gchar *GetVaildModelFile () {
+    gchar *model_file;
+
+    /* nnfw needs a directory with model file and metadata in that directory */
+    g_autofree gchar *model_path = g_build_filename (root_path, "tests", "test_models", "models", NULL);
+
+    g_autofree gchar *meta_file = g_build_filename (model_path, "metadata", "MANIFEST", NULL);
+    if (!g_file_test (meta_file, G_FILE_TEST_EXISTS)) {
+      return NULL;
+    }
+
+    model_file = g_build_filename (model_path, "add.tflite", NULL);
+    if (!g_file_test (model_file, G_FILE_TEST_EXISTS)) {
+      g_free (model_file);
+      return NULL;
+    }
+    return model_file;
   }
 
-  model_file = g_build_filename (model_path, "add.tflite", NULL);
-  g_free (meta_file);
-  g_free (model_path);
+protected:
+  /**
+   * @brief Construct a new MLAPIInferenceNNFW object
+   */
+  MLAPIInferenceNNFW ()
+    : single_h(nullptr), pipeline_h(nullptr), input(nullptr), input2(nullptr),
+      output(nullptr), root_path(nullptr), valid_model(nullptr)
+  {
```

Comment:
```
  ml_tensors_info_h in_info, out_info;
  ml_tensors_info_h in_res, out_res;
  ml_tensor_dimension in_dim, out_dim, res_dim;
```
init variables to prevent svace issue.

Reply:
OK. I will fix it using constructor initialization as below.

```cpp
  MLAPIInferenceNNFW ()
    : single_h(nullptr), pipeline_h(nullptr), in_info(nullptr), out_info(nullptr),
      in_res(nullptr), out_res(nullptr), in_dim{1, 1, 1, 1}, out_dim{1, 1, 1, 1}, res_dim{1, 1, 1, 1},
      input(nullptr), input2(nullptr), output(nullptr), root_path(nullptr), valid_model(nullptr) {  }
```

Reply:
I found out that the below constructor initialization can cause the `Doxygen Tag` issue because of `{`.
```
in_dim{1, 1, 1, 1}, out_dim{1, 1, 1, 1}, res_dim{1, 1, 1, 1}
```

#### Error message
```
[DEBUG] The current file name is (tests/capi/unittest_capi_inference_nnfw_runtime.cc). 
[DEBUG] ( tests/capi/unittest_capi_inference_nnfw_runtime.cc ) file is source code with the text format.
[ERROR] File name: tests/capi/unittest_capi_inference_nnfw_runtime.cc, 66 line, input(nullptr), function needs @brief tag 
[ERROR] Failed. Doxygen documentation.
```
In my opinion, it looks like a script bug. So I will set them in the constructor body.

Reply:
for ml_tensor_dimension (ML_TENSOR_RANK_LIMIT) : in_dim, out_dim, res_dim
how about adding init code in constructor? (Now dimension zero-init in _ml_tensors_info_initialize)
```
for (i = 0; i < ML_TENSOR_RANK_LIMIT; i++)
  in_dim[i] = out_dim[i] = res_dim[i] = 0;
```

Reply:
The code you recommended is more elegant! I will apply this! :+1: